python/ray/data/_internal/stats.py

```diff
@@ -452,6 +452,9 @@ def get_total_cpu_time(self) -> float:
     def get_max_heap_memory(self) -> float:
         parent_memory = [p.get_max_heap_memory() for p in self.parents]
         parent_max = max(parent_memory) if parent_memory else 0
+        if not self.stages_stats:
+            return parent_max
+
```

Comment:
Could you help me understand this change? I'm not sure how it relates to this PR.

Reply:
Ah it was a separate bug that I came across when I was writing the tests. We hit an error when trying to take the `max` in the line below, if self.stage_stats is empty.